main.py

```diff
@@ -0,0 +1,35 @@
+
+from discord.ext import commands
+import discord, datetime, asyncio, json
+
+with open("settings.json", "r") as read_file: data = json.load(read_file)
+REPORT_TIME = data["time"]
+CHANNEL_ID = int(data["channel_id"])
+MESSAGE = data["message"]
+TOKEN = open("token.txt", "r").read()
+
+intents = discord.Intents.all()
+client = commands.Bot(command_prefix='!', intents=intents)
+client.remove_command('help')
+
+@client.event
+async def on_ready():
+    client.loop.create_task(timer())
+    print(f'{client.user} online!')
+
+async def timer():
+    while True:
+        await asyncio.sleep(1)
```

Comment:
Co dát ten sleep na celou minutu ať to bere míň cpu?

Reply:
A nebo mně ještě napadlo Github podporuje CRON github akce, takže by jsme mohli udělat Github akci, která spustí nějaký python script, a nemusí to vůbec běžet na serveru. https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#schedule

Reply:
> A nebo mně ještě napadlo Github podporuje CRON github akce, takže by jsme mohli udělat Github akci, která spustí nějaký python script, a nemusí to vůbec běžet na serveru. https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#schedule

Existuje pravidlo: 

Držte kód, který k sobe patří u sebe.

Otázka:
Co když změníte GitHub na GitLab? Co pak? Bot se rozbije? - Lepší to je držet vše na jednom místě, takže přístup skrz async v .py je v pořádku. Jen trochu doladit ten sleep (viz. můj návrh). :)

Reply:
@Venca321 
Dodatek k predchozimu navrhu - discordi python module dokonce ma uz reseni primo pro tenhle typ problemu... skrz "tasks" modul, viz:

https://discordpy.readthedocs.io/en/stable/ext/tasks/index.html